Framework/PythonInterface/test/python/plugins/algorithms/SaveReflectionsTest.py

```diff
@@ -0,0 +1,174 @@
+            column_data.append(fractional_peaks.column(name))
+
+        # Insert the addtional HKL columns into the data
+        for i, col in enumerate(hklm.T.tolist()):
+            column_data.insert(i + 2, col)
+
+        # Insert the columns into the table workspace
+        for i in range(fractional_peaks.rowCount()):
+            row = [column_data[j][i] for j in range(indexed.columnCount())]
+            indexed.addRow(row)
+
+        return indexed
+
+    def _get_reference_result(self, name):
+        path = FileFinder.getFullPath(name)
+        if path is None or path == "":
+            raise RuntimeError("Could not find unit test data: {}".format(name))
+        return path
+
+    def test_save_fullprof_format(self):
+        # Arrange
+        reference_result = self._get_reference_result("fullprof_format.hkl")
+        file_name = os.path.join(self._test_dir, "test_fullprof.hkl")
+        output_format = "Fullprof"
+
+        # Act
+        SaveReflections(InputWorkspace=self._workspace, Filename=file_name, Format=output_format)
+
+        # Assert
+        self.assertTrue(compare_file(reference_result, file_name))
+
+    def test_save_fullprof_format_modulated(self):
+        # Arrange
+        workspace = self._create_modulated_peak_table()
+        reference_result = self._get_reference_result("fullprof_format_modulated.hkl")
+        file_name = os.path.join(self._test_dir, "test_fullprof_modulated.hkl")
+        output_format = "Fullprof"
+
+        # Act
+        SaveReflections(InputWorkspace=workspace, Filename=file_name, Format=output_format)
+
+        # Assert
+        self.assertTrue(compare_file(reference_result, file_name))
+
+    def test_save_jana_format(self):
+        # Arrange
+        reference_result = self._get_reference_result("jana_format.hkl")
+        file_name = os.path.join(self._test_dir, "test_jana.hkl")
+        output_format = "Jana"
+
+        # Act
+        SaveReflections(InputWorkspace=self._workspace, Filename=file_name, Format=output_format)
+
+        # Assert
+        self.assertTrue(compare_file(reference_result, file_name))
+
+    def test_save_GSAS_format(self):
+        # Arrange
+        reference_result = self._get_reference_result("gsas_format.hkl")
+        file_name = os.path.join(self._test_dir, "test_gsas.hkl")
+        output_format = "GSAS"
+
+        # Act
+        SaveReflections(InputWorkspace=self._workspace, Filename=file_name, Format=output_format)
+
+        # Assert
+        self.assertTrue(compare_file(reference_result, file_name))
+
+    def test_save_SHELX_format(self):
+        # Arrange
+        reference_result = self._get_reference_result("shelx_format.hkl")
+        file_name = os.path.join(self._test_dir, "test_shelx.hkl")
+        output_format = "SHELX"
+
+        # Act
+        SaveReflections(InputWorkspace=self._workspace, Filename=file_name, Format=output_format)
+
+        # Assert
+        self.assertTrue(compare_file(reference_result, file_name))
```

Comment:
You should also add tests for saving the modulated Jana and GSAS/SHELX workspaces (i.e. check the error is raised).

Reply:
Will do.